
syntax = "proto2";

message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressReply{
  optional string message = 1;
}

message DisplayAll {
    optional string display1 = 1;
}

message AddressBook {
  repeated Person person = 1;
}

service Greeter{
    rpc InsertEntry (Person) returns (AddressReply) {};
    rpc AllEntries (DisplayAll) returns (AddressBook) {};
}


